eth/api.go

```diff
@@ -812,3 +813,23 @@ func (api *ScrollAPI) GetSkippedTransactionHashes(ctx context.Context, from uint
 
 	return hashes, nil
 }
+
+// CalculateRowConsumptionByBlockNumber
+func (api *ScrollAPI) CalculateRowConsumptionByBlockNumber(ctx context.Context, number rpc.BlockNumber) (*types.RowConsumption, error) {
+	block := api.eth.blockchain.GetBlockByNumber(uint64(number.Int64()))
+	if block == nil {
+		return nil, errors.New("block not found")
+	}
+
+	// todo: fix temp AsyncChecker leaking the internal Checker instances
```

Comment:
this is due to Rust wrappers providing no way of freeing a checker once allocated. Will fix during write of the rust glue code that we have in l2geth.